internal/build/job/maker.go

```diff
@@ -97,32 +99,39 @@ func (m *maker) MakeJob(mod kmmv1beta1.Module, buildConfig *kmmv1beta1.Build, ta
 		kanikoImageTag = buildConfig.KanikoParams.Tag
 	}
 
+	specTemplate := v1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{"Dockerfile": buildConfig.Dockerfile},
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Args:         args,
+					Name:         "kaniko",
+					Image:        "gcr.io/kaniko-project/executor:" + kanikoImageTag,
+					VolumeMounts: volumeMounts,
+				},
+			},
+			NodeSelector:  mod.Spec.Selector,
+			RestartPolicy: v1.RestartPolicyOnFailure,
+			Volumes:       volumes,
+		},
+	}
+	specTemplateHash, err := hashstructure.Hash(specTemplate, nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not hash job's spec template: %v", err)
+	}
+
 	job := &batchv1.Job{
 		ObjectMeta: metav1.ObjectMeta{
 			GenerateName: mod.Name + "-build-",
 			Namespace:    mod.Namespace,
 			Labels:       labels(mod, targetKernel),
+			Annotations:  map[string]string{jobHashAnnotation: fmt.Sprintf("%d", specTemplateHash)},
```

Comment:
Is `fmt.Sprintf("%d", specTemplateHash)` working?
I tried `fmt.Println(string(specTemplateHash))` and I got an empty value.